File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <ctype.h>

// Declaring Functions
string input(void);
int letters(string s, int n);
int words(string s, int n);
int sentences(string s, int n);
float index(float n_l, float n_w, float n_s);
void output(float n);

// Main Function
int main(void)
{
    string s = input();
    int n = strlen(s);
    int n_l = letters(s, n);
    int n_w = words(s, n);
    int n_s = sentences(s, n);
    float i = index(n_l, n_w, n_s);
    printf("%i letters\n", n_l);
    printf("%i words\n", n_w);
    printf("%i sentences\n", n_s);
    output(i);
}

// Taking input
string input(void)
{
    return get_string("Enter your text: \n");
}

// Counting letters
int letters(string s, int n)
{
    int n1 = 0;
    for (int i = 0; i < n; i++)
    {
        if (isalpha(s[i]))
        {
            n1++;
        }
    }
    return n1;
}

// Counting words
int words(string s, int n)
{
    int n1 = 1;
    for (int i = 0; i < n; i++)
    {
        if (s[i] == ' ')
        {
            n1++;
        }
    }
    return n1;
}

// Counting sentences
int sentences(string s, int n)
{
    int n1 = 0;
    for (int i = 0; i < n; i++)
    {
        if (s[i] == '.' || s[i] == '?' || s[i] == '!')
        {
            n1++;
        }
    }
    return n1;
}

// Computing index
float index(float n_l, float n_w, float n_s)
{
    float l = (n_l / n_w) * 100.0;
    float s = (n_s / n_w) * 100.0;
    float i = (0.0588 * l) - (0.296 * s) - (15.8);
    return i;
}

// Printing output
void output(float n)
{
    int i = round(n);
    if (i >= 1 && i <= 16)
    {
        printf("Grade %i\n", i);
    }
    else if (i > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }
}

File: sentimental-readability/readability.py
from cs50 import get_string

# Main function


def main():
    text = get_input()
    letters = letter(text)
    words = word(text)
    sentences = sentence(text)
    index = ind(letters, words, sentences)
    output(index)

# Getting input, text, from user


def get_input():
    return get_string("Text: ")

# Counting Letters


def letter(text):
    count = 0
    for c in text:
        if c.isalpha():
            count += 1
    return count

# Counting Words


def word(text):
    count = 1
    for c in text:
        if c == " ":
            count += 1
    return count

# Counting Sentences


def sentence(text):
    count = 0
    for c in text:
        if c == "." or c == "?" or c == "!":
            count += 1
    return count

# Calculating Index


def ind(letters, words, sentences):
    index = (5.88 * (letters / words)) - (29.6 * (sentences / words)) - (15.8)
    return round(index)

# Printing Output


def output(index):
    if index >= 1 and index <= 16:
        print(f"Grade {index}")
    elif index > 16:
        print("Grade 16+")
    else:
        print("Before Grade 1")


# Calling main function
main()